Storage/RecentlyClosedTabs.swift

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public class ClosedTabsStore {
+    let prefs: Prefs
+
+    lazy public var tabs: [ClosedTab] = {
+        guard let tabsArray = self.prefs.objectForKey("recentlyClosedTabs"),
+              let unarchivedArray = NSKeyedUnarchiver.unarchiveObjectWithData(tabsArray as! NSData) as? [ClosedTab] else {
+            return []
+        }
+        return unarchivedArray
+    }()
+
+    public init(prefs: Prefs) {
+        self.prefs = prefs
+    }
+
+    public func addTab(url: NSURL, title: String?, faviconURL: String?) {
+        let recentlyClosedTab = ClosedTab(url: url, title: title ?? "", faviconURL: faviconURL ?? "")
+        tabs.insert(recentlyClosedTab, atIndex: 0)
```

Comment:
Just noticed that this array is unbounded, meaning the tabs list can grow infinitely long. Set a limit so that we don't store more than N.